alphastats/tl/differential_expression_analysis.py

```diff
@@ -0,0 +1,345 @@
+            group1 = parameters[DeaParameters.GROUP1]
+            group2 = parameters[DeaParameters.GROUP2]
+        else:
+            grouping_column = parameters[DeaParameters.METADATA][
+                parameters[DeaParameters.GROUPING_COLUMN]
+            ]
+            sample_column = parameters[DeaParameters.METADATA][Cols.SAMPLE]
+            group1 = [
+                sample
+                for sample, group in zip(sample_column, grouping_column)
+                if group == parameters[DeaParameters.GROUP1]
+            ]
+            group2 = [
+                sample
+                for sample, group in zip(sample_column, grouping_column)
+                if group == parameters[DeaParameters.GROUP2]
+            ]
+        return group1, group2
+
+
+class DifferentialExpressionAnalysisTTest(DifferentialExpressionAnalysisTwoGroups):
+    """This class implements the t-test differential expression analysis."""
+
+    @staticmethod
+    def _allowed_parameters() -> List[str]:
+        return [
+            DeaParameters.TEST_FUN,
+            DeaParameters.FDR_METHOD,
+            DeaParameters.GROUP1,
+            DeaParameters.GROUP2,
+            DeaParameters.GROUPING_COLUMN,
+            DeaParameters.METADATA,
+            PreprocessingStateKeys.LOG2_TRANSFORMED,
+        ]
+
+    def _extend_validation(self, parameters: dict):
+        """Validates the input and parameters for the t-test differential expression analysis.
+
+        This function checks for the required parameters for the t-test analysis, namely test_fun and fdr_method. The test_fun must be either scipy.stats.ttest_ind or scipy.stats.ttest_rel and the fdr_method must be one of 'bh' or 'by'.
+
+        Parameters:
+        parameters (dict): The parameters for the analysis.
+        """
+        super()._extend_validation(parameters)
+        if parameters["test_fun"] not in [
+            "independent",
+            "paired",
+        ]:
+            raise ValueError(
+                "test_fun must be either 'independent' for scipy.stats.ttest_ind or 'paired' for scipy.stats.ttest_rel."
+            )
+        if parameters["fdr_method"] not in ["fdr_bh", "bonferroni"]:
+            raise ValueError("fdr_method must be one of 'fdr_bh', 'bonferroni'.")
+
+    def _run_statistical_test(self, **kwargs) -> pd.DataFrame:
+        """Runs the t-test analysis and returns the result.
+        Wrapper to method with actual method parameters and implementation.
+
+        Returns:
+        pd.DataFrame: The result of the analysis.
+        """
+        group1, group2 = self._get_group_members(kwargs)
+        result = self._statistical_test_fun(
+            input_data=self.input_data,
+            group1=group1,
+            group2=group2,
+            is_log2_transformed=kwargs[PreprocessingStateKeys.LOG2_TRANSFORMED],
+            test_fun=kwargs[DeaParameters.TEST_FUN],
+            fdr_method=kwargs[DeaParameters.FDR_METHOD],
```

Comment:
I would strongly prefer explicit parameters here

Reply:
Done throughout. Wherever I can predict the arguments they are now explicit. This removed many tests and validations as well.